packages/ubuntu_bootstrap/lib/services/installer_service.dart

```diff
@@ -33,16 +36,28 @@ class InstallerService {
 
   Future<void> load() async {
     await monitorStatus().firstWhere((s) => s?.isLoading == false);
-    _pages ??=
-        await _client.getInteractiveSections() ?? await _config?.getPages();
+    _subiquityPages = (await _client.getInteractiveSections())?.toSet();
+    _excludedPages = _pageConfig?.excludedPages;
+
+    if ((_subiquityPages?.isNotEmpty ?? false) &&
+        (_excludedPages?.isNotEmpty ?? false)) {
```

Comment:
Since they are sets you could just do the intersection of the sets and then loops over that